Feature add fwversion to manifest json (#5679)
nrfconnect/sdk-nrf: doc/nrf/libraries/networking/nrf_cloud.rst

====================
The nRF Connect SDK build system places output images in ``<build folder>/zephyr``.

If :kconfig:`CONFIG_BOOTLOADER_MCUBOOT` is set, the build system creates the file ``dfu_application.zip`` containing ``app_update.bin`` and ``manifest.json``.
Further, if :kconfig:`CONFIG_IMG_MANAGER` and :kconfig:`CONFIG_MCUBOOT_IMG_MANAGER` are also set, then the application will be able to process FOTA updates.

The file ``app_update.bin`` is a signed version of your application.  The signature matches what MCUboot expects, so it will allow this file to be used as an update.
The build system creates ``manifest.json`` using information in the ``zephyr.meta`` output file.  This includes the Zephyr and |NCS| git hashes for the commits used to build the application.
If your working tree contains uncommitted changes, the build system adds the suffix ``-dirty`` to the relevant version field.

When you use the zip file to create an update bundle, the `nRF Cloud`_ UI populates the ``Name`` and ``Version`` fields from manifest.json. However, you are free to change them as needed.
The UI populates the ``Version`` field from only the |NCS| version field in ``manifest.json``.

> Alternatively, you can use the file ``app_update.bin`` to create an update bundle, but you will need to enter the ``Name`` and ``Version`` fields manually.
> See `nRF Cloud Getting Started FOTA documentation` to learn how to create an update bundle.

Reviewer: ```suggestion
Alternatively, you can use the :file:`app_update.bin` file to create an update bundle, but you need to enter the ``Name`` and ``Version`` fields manually.
See `nRF Cloud Getting Started FOTA documentation`_ to learn how to create an update bundle.
```

The message payload of the :c:enumerator:`NRF_CLOUD_EVT_FOTA_DONE` event contains the :c:enum:`nrf_cloud_fota_type` value.
If the value equals :c:enumerator:`NRF_CLOUD_FOTA_MODEM`, the application can optionally avoid a reboot by performing reinitialization of the modem and calling the :c:func:`nrf_cloud_modem_fota_completed` function.

> Building FOTA images

Reviewer: This is rather generic stuff (except for the Cloud part of it) and I guess it could be better placed in, for example in the Getting started/Building and programming an application section (https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/gs_programming.html). The generic part could then be included here from the Building and programming an application. Opinions?

> Building FOTA images

Reviewer: This is rather generic stuff (except for the Cloud part of it) and I guess it could be better placed in, for example in the Getting started/Building and programming an application section (https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/gs_programming.html). Or then the Application development/Build and configuration system (https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/app_build_system.html). The generic part could then be included here from the place where this was added. Now, the changelog entry is placed under the Application development section, although this is documented only in this nRF Cloud library doc. If you decide to have this section only in this library doc, please move the changelog entry to the proper place under Libraries/Libraries for networking.

Reviewer: My colleague also commented offline: "Rather than misplaced, it's hard to understand if nRF Cloud FOTA is using any of the FOTA libraries from the NCS, like FOTA download, or the Bootloader/DFU processes."

Author: @peknis I do not think the "generic stuff" belongs under Libraries for networking.  The changes I made in the PR were not specific to nRF Cloud, but were certainly relevant to it.  The changes are in general application build system locations in the source tree (/cmake, /modules/mcuboot, /scripts/bootloader), so the changelog entry should be where I put it, in my opinion.

I'm fine with moving the first three paragraphs to the second link you mention and simply reference that here.

Regarding what dependencies the nRF Cloud FOTA library has on other FOTA libraries, I agree it should be documented.  The file subsys/net/lib/nrf_cloud/Kconfig.nrf_cloud_fota forces on a number of libraries or features on.

====================
The |NCS| build system places output images in the :file:`<build folder>/zephyr` folder.

> If :kconfig:`CONFIG_BOOTLOADER_MCUBOOT` is set, the build system creates the file :file:`dfu_application.zip` file containing files :file:`app_update.bin` and :file:`manifest.json`.
> Further, if :kconfig:`CONFIG_IMG_MANAGER` and :kconfig:`CONFIG_MCUBOOT_IMG_MANAGER` are also set, then the application will be able to process FOTA updates.

Reviewer: ```suggestion
If  the :kconfig:`CONFIG_BOOTLOADER_MCUBOOT` option is set, the build system creates the :file:`dfu_application.zip` file containing the :file:`app_update.bin` and :file:`manifest.json` files.
Further, if the options :kconfig:`CONFIG_IMG_MANAGER` and :kconfig:`CONFIG_MCUBOOT_IMG_MANAGER` are also set, the application will be able to process FOTA updates.
```

The :file:`app_update.bin` file is a signed version of your application.

> The signature matches to what MCUboot expects, so it allows this file to be used as an update.

Reviewer: ```suggestion
The signature matches to what MCUboot expects and allows this file to be used as an update.
```

The build system creates a :file:`manifest.json` file using information in the :file:`zephyr.meta` output file.
This includes the Zephyr and |NCS| git hashes for the commits used to build the application.
If your working tree contains uncommitted changes, the build system adds the suffix ``-dirty`` to the relevant version field.

When you use the zip file to create an update bundle, the `nRF Cloud`_ UI populates the ``Name`` and ``Version`` fields from the :file:`manifest.json` file.
However, you are free to change them as needed.

> The UI populates the ``Version`` field from only the |NCS| version field in :file:`manifest.json` file.

Reviewer: ```suggestion
The UI populates the ``Version`` field from only the |NCS| version field in the :file:`manifest.json` file.
```
